modules/meshrenderinggl/src/processors/rasterizationrenderer.cpp

```diff
@@ -0,0 +1,180 @@
+#include <inviwo/core/util/stdextensions.h>
+#include <modules/opengl/openglutils.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/shader/shaderutils.h>
+#include <modules/base/algorithm/dataminmax.h>
+#include <modules/opengl/image/layergl.h>
+#include <modules/opengl/openglcapabilities.h>
+#include <modules/opengl/rendering/meshdrawergl.h>
+
+#include <sstream>
+#include <chrono>
+#include <variant>
+
+#include <fmt/format.h>
+
+namespace inviwo {
+namespace {
+void configComposite(BoolCompositeProperty& comp) {
+
+    auto callback = [&comp]() mutable {
+        comp.setCollapsed(!comp.isChecked());
+        for (auto p : comp) {
+            if (p == comp.getBoolProperty()) continue;
+            p->setReadOnly(!comp.isChecked());
+        }
+    };
+
+    comp.getBoolProperty()->onChange(callback);
+    callback();
+}
+}  // namespace
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo RasterizationRenderer::processorInfo_{
+    "org.inviwo.RasterizationRenderer",  // Class identifier
+    "Rasterization Renderer",            // Display name
+    "Mesh Rendering",                    // Category
+    CodeState::Experimental,             // Code state
+    Tags::GL,                            // Tags
+};
+const ProcessorInfo RasterizationRenderer::getProcessorInfo() const { return processorInfo_; }
+
+RasterizationRenderer::RasterizationRenderer()
+    : Processor()
+    , rasterizations_("rastarizations")
+    , imageInport_(std::make_shared<ImageInport>("imageInport"))
+    , outport_("image")
+    , forceOpaque_("forceOpaque", "Shade Opaque", false, InvalidationLevel::InvalidResources)
+    , flr_(imageInport_) {
+
+    // query OpenGL Capability
+    supportsFragmentLists_ = FragmentListRenderer::supportsFragmentLists();
+    supportesIllustration_ = FragmentListRenderer::supportsIllustration();
+    if (!supportsFragmentLists_) {
+        LogProcessorWarn(
+            "Fragment lists are not supported by the hardware -> use blending without sorting, may "
+            "lead to errors");
+    }
+    if (!supportesIllustration_) {
+        LogProcessorWarn(
+            "Illustration Buffer not supported by the hardware, screen-space silhouettes not "
+            "available");
+    }
+
+    // input and output ports
+    addPort(rasterizations_);
+    addPort(*imageInport_).setOptional(true);
+    addPort(outport_);
+
+    addProperties(forceOpaque_, illustrationSettings_.enabled_);
+
+    illustrationSettings_.enabled_.readonlyDependsOn(
+        forceOpaque_, [this](const auto& prop) { return !supportesIllustration_ && prop.get(); });
+
+    flrReload_ = flr_.onReload([this]() { invalidate(InvalidationLevel::InvalidResources); });
+}
+
+RasterizationRenderer::IllustrationSettings::IllustrationSettings()
+    : enabled_("illustration", "Use Illustration Effects", false)
+    , edgeColor_("edgeColor", "Edge Color", vec3(0.f, 0.f, 0.f))

```

Comment:
Could use the new constraints now?
nice to have, not need to...

Reply:
use 
`edgeColor_("edgeColor", "Edge Color", ordinal::color(vec3(0.0f)))`

Reply:
Nice, if time admits it could be worthwhile to go through each property and see if constraints should be added